app/controllers/application_controller.rb

```diff
@@ -58,7 +58,7 @@ def decorated_user
   end
 
   def after_sign_in_path_for(user)
-    stored_location_for(user) || session[:saml_request_url] || profile_path
+    stored_location_for(user) || profile_path
```

Comment:
what I've learned from looking through the code is that we use the presence of `saml_request_url` to know if there is a service provider involved for a request. We do some logic around that. I think we can refactor code in follow up PR to remove all references to `saml_request_url` since we should know if there is a service provider based on presence of `session[:sp]`